### DISCUSSION TITLE
MQTT Befehle

### DISCUSSION THREAD

**Discussion by neolauren-cmd** [hoylabs/OpenDTU-OnBattery#400] [General]
Hallo zusammen, 

ich möchte gerne per MQTT Befehl  meinen Dynamic Power Limiter aktivieren bzw. deaktivieren.
Ist das irgendwie möglich  bzw. wie müsste der Befehl aussehen?
solar/cmd/.....?

Danke für einen Tipp.

Gruß 
Neo

**Comment by sylvio69** [comment]
solar/powerlimiter/cmd/mode 
0 - Normal Mode, 1 - PL Disable, 2 - PL Solar Passthrough

**Comment by schlimmchen** [comment]
sylvio69 liegt richtig, der Vollständigkeit halber ist hier die Quelle dieser Info: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/MQTT

@neolauren-cmd Bitte diese Diskussion schließen und eine Antwort auswählen.

**Reply by neolauren-cmd** [reply]
Danke Euch.
Wenn ich aber in Node_red das hier eingebe:

solar/powerlimiter/cmd/mode/0
oder
solar/powerlimiter/cmd/mode/1
oder
solar/powerlimiter/cmd/mode/2

passiert nicht in der WebIF. Oder habe ich falsch eingegeben?

Gruß

**Reply by schlimmchen** [reply]
Wo im Web Interface? An der Seite mit den Einstellungen ändert sich nichts. Hast du VE.Direct eingeschaltet? Dann sollte das Icon des DPL dir sagen, dass er aus ist. Andernfalls kannst du das Verhalten des DPL beobachten, da kann man zwischen den Modi ja Unterschiede feststellen (eingestelltes Limit am Inverter).

**Reply by sylvio69** [reply]
Ob sich das in der GUI widerspiegelt bin ich mir nicht sicher. Ich glaube das Icon hat sich nicht geändert bei mir. Ist aber schon ewig her als ich das letzte mal geschaut hatte. Ansonsten, Du musst 0, 1 oder 2 als Payload übergeben. Du hast jetzt ggf. drei neue Topic's erzeugt. Schau doch mal mit dem MQTT Explorer rein.

**Reply by sylvio69** [reply]
Gerade nochmal getestet. Das Icon ändert sich tatsächlich. Also ist in der GUI doch nachvollziehbar.

**Reply by neolauren-cmd** [reply]
Funktioniert mit dem Payload senden funktionert und ich kann es im MQQT Explorer nachvollziehen. In der GUI ändert es sich bei mir leider nicht. Zeigt immer aktiviert an.

**Reply by spcqike** [reply]
Kann es sein dass du den payload nicht korrekt sendest? Sollte es nicht 

solar/powerlimiter/cmd/mode {payload} 

sein? 

solar/powerlimiter/cmd/mode/0 wäre ja ein ganz anderer topic ohne payload.

**Reply by neolauren-cmd** [reply]
Ja das hatte ich am Anfang. Sende jetzt mit

solar/powerlimiter/cmd/mode

den entsprechenden Payload 0 oder 1. Die Anzeige im Powerlimiter beleibt aber immer aktiviert, egal ob 0 oder 1.

**Reply by sylvio69** [reply]
Ändert sich der Status entsprechend dem was Du schickst (solar/powerlimiter/status)?

**Reply by neolauren-cmd** [reply]
ja der ändert sich entsprechend 0 oder 1, aber komischerweise nicht visuell. Komisch, dass es bei dir funktioniert.

**Reply by neolauren-cmd** [reply]
Kann jemand mal probieren den Modus 2 zu schreiben. Bei mir springt der WR immer auf Modus 1 (obwohl ich 2 schreibe) und dann bleibt der WR aus.

**Reply by schlimmchen** [reply]
Funktioniert bei mir wunderbar. Sowohl vom Zustand 1 als auch vom Zustand 0 kommend.

Hast du denn einen Aufbau mit funktionierendem VE.Direct?

**Reply by neolauren-cmd** [reply]
Nein den habe ich nicht. Was ist denn VE.direct? Ich nutzt die Firmware für eine Nulleinspeisung ohne Akku. 

Das heisst du siehst dann im MQQT Explorer den Modus 2?
Wenn ich die 2 schreibe erscheint die 1 im Status. Mit der 0 klappt es.

**Reply by schlimmchen** [reply]
Okay. Also jetzt hab ich dein Problem verstanden: Du bemängelst, dass `solar/powerlimiter/status/mode` nicht 2 ist, wenn du 2 in `solar/powerlimiter/cmd/mode` veröffentlicht hast. Das ist ein valider Punkt und du hast Recht, das funktioniert nicht wie erwartet, habe ich gerade nachvollzogen. Dennoch funktioniert der DPL wie durch den Wert 2 vorgegeben.

Darüber hinaus kannst du natürlich nicht erwarten, dass der DPL Unconditional-Solar-Passthrough macht, denn er kennt ja deinen solaren Ertrag nicht ohne VE.Direct. Jedenfalls ist es richtig, dass der DPL deinen Inverter dann ausschaltet und ausgeschalten hält.

**Reply by neolauren-cmd** [reply]
Meine Intension war das ich per MQTT Befehl den Power Limiter einschalten kann. Dann funktioniert die Nulleinspeisung, wie in dem Untermenü eingestellt ist. 

Und dann wollte ich per MQTT Befehl den Power Limiter ausschalten. Da sollte der WR die Leitung bringen, die ich ihm vorgebe, ohne Regelung halt.

Aber anscheinend funktioniert das nicht so.

**Reply by schlimmchen** [reply]
> Da sollte der WR die Leitung bringen, die ich ihm vorgebe

Dieser Wert muss dann irgendwo herkommen, z.B. durch MQTT. Und dann tut der WR das auch. Ich versteh dein Problem nicht. Was du beschreibst ist möglich:

* Stelle den DPL der OpenDTU-OnBattery so ein, dass die Nulleinspeisung macht.
* Setze per MQTT den DPL-Modus "normal" (0) um Nulleinspeisung zu haben, der durch OpenDTU-OnBattery geregelt wird.
* Setze per MQTT den DPL-Modus "off" (1) um den DPL auszuhebeln. Steuere dann die Leistung des WR per MQTT.

**Reply by neolauren-cmd** [reply]
Ah okay. Das probiere ich gleich mal noch.

**Reply by neolauren-cmd** [reply]
Denke es funktioniert. Im Moment hat der WR zwecks Gewitter abgeschaltet. 

Danke füre eure Unterstützung.

**Reply by neolauren-cmd** [reply]
Konnte jetzt probieren und es klappt so wie beschrieben. Das einzige war, nachdem ich den DPL deaktiviert habe und das Limit gesetzt habe, ist der WR auch ausgeblieben. Ich musste ihn per WebIf einschalten, dann hat es geklappt. 
Daher die Frage, ist das normal? Gibt es auch einen MQTT Befehl, damit ich den WR wieder einschalten kann?

**Reply by neolauren-cmd** [reply]
Habs gefunden :-)

**Comment by neolauren-cmd** [comment]
Merci

**Comment by Webranger2** [comment]
> Habs gefunden :-)

Was war die lösung?

**Comment by Webranger2** [comment]
> sylvio69 liegt richtig, der Vollständigkeit halber ist hier die Quelle dieser Info: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/MQTT

Wenn ich das per mqtt setzte seh ich das aber nicht in der Webobfläche oder? Wenn ich da eine 1 setzte für "PL Disable" Bleibt der Knopf weiterhin aktiv! ist das richtig und so gewollt?
<img width="1816" alt="Bildschirmfoto 2024-01-14 um 17 32 09" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/44174579/febe6058-7ac4-4ba3-83b3-72acce282983">

**Reply by schlimmchen** [reply]
Ja, das ist richtig so.

Der DPL bräuchte eine eigene Kachel in der Live-Ansicht, damit man insb. den Wert dieses sonst "versteckten" Modus nachvollziehen kann. In den Einstellungen wird der DPL aber aktiv bleiben, insbesondere weil er ja seine Arbeit wieder aufnehmen soll, wenn man per MQTT wieder "normal operation" wünscht.
